### ISSUE TITLE
addWorkingTime / subtractWorkingTime issues on leap year

### ISSUE THREAD

**Issue by anders-lassen** [lennym/moment-business-time#39]
## Errors
`moment("2020-01-06 13:00").addWorkingTime(1, "seconds").format()`
returns: `"2020-01-31T13:00:01+01:00"`

`moment("2020-01-06 13:00").subtractWorkingTime(1, "seconds").format()`
returns: `"2020-01-31T12:59:59+01:00"`

hint: look at the day it went from 06 to 31

I've tested the above with dates in 2024 as well, as it is also a leap year. Still returns falsy.

## Configs:
```javascript
_workinghours: {
    0: (2) ["", ""]
    1: (2) ["07:00", "15:00"]
    2: (2) ["07:00", "15:00"]
    3: (2) ["07:00", "15:00"]
    4: (2) ["07:00", "15:00"]
    5: (2) ["07:00", "13:00"]
    6: (2) ["", ""]
}
```

## The fix
```javascript
function addOrSubtractMethod(fn) {
	     return function (num, unit) {
		 if (typeof unit !== 'string') {
		     throw new Error('unit must be defined');
		 }
		 if (typeof num !== 'number') {
		     throw new Error('duration must be defined');
		 }
		 var args = [].slice.call(arguments);
		 if (args.length % 2) {
		     throw new Error('moment#(add/subtract)WorkingTime requires an even number of arguments');
		 }

		 var d = this.clone(); // Originally this didn't include .clone(), but somehow because of leap year the copy call (further down) returned the above error. Why I haven't figured out

		 while (args.length >= 2) {
		     d = fn.bind(null, d).apply(null, args.slice(-2));
		     args = args.slice(0, -2);
		 }

		 return copy(d, this); // This returned the wrong date
	     };
	 }
```

**Comment by lennym** [comment]
Thanks for opening an issue. I'll take a look at it.

**Comment by lennym** [comment]
It's been a while since I looked at this, and so I updated some of the dependencies, and that seems to resolve the issue.

I added [a unit test with your test case](https://github.com/lennym/moment-business-time/blob/master/test/spec.business-hours.js#L185-L187) to be sure.

Anyway, it's published at v0.8.0 so let me know if you have any more issues.
